Sources/Jetworking/Client/ResponseHandler.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+
+enum ResponseHandler {
+    typealias WrappedCompletion<ResponseType> = (HTTPURLResponse, Data?, Decoder, @escaping Client.RequestCompletion<ResponseType>) -> () -> Void
+    enum CompletionWrapper {
```

Comment:
Do we need the `CompletionWrapper` enum?

Reply:
I wanted to emphasise that the following methods can be used to obtain a `WrappedCompletion`. One can argue if it's a subtype of `ResponseHandler`.  Since it's only useful in the `ResponseHandler` context (one of the `CompletionWrapper` should be used to populate the `completionWrapper` parameter of the `evaluate` function) I would keep it as a subtype.

Reply:
I understand your point but there is no need to create an embedded type `CompletionWrapper` since it is not really a type. It is just as substructure and that is superfluous since it is more difficult to know that you have to chain multiple types to create a completion handler -> `ResponseHandler.CompletionWrapper.void`. It would be. much easier to use the factory function directly: `ResponseHandler.makeVoidWrapper`.

Reply:
These methods are now renamed